File: module_news/src/main/java/com/fly/tour/news/NewsDetailActivity.java
package com.fly.tour.news;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProvider;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.Nullable;
import android.widget.TextView;
import com.fly.tour.api.news.NewsDetail;
import com.fly.tour.common.mvvm.BaseMvvmActivity;
import com.fly.tour.common.event.KeyCode;
import com.fly.tour.news.mvvm.factory.NewsViewModelFactory;
import com.fly.tour.news.mvvm.viewmodel.NewsDetailViewModel;

public class NewsDetailActivity extends BaseMvvmActivity<NewsDetailViewModel>{
    public static void startNewsDetailActivity(Context context,int id){
        Intent intent = new Intent(context, NewsDetailActivity.class);
        intent.putExtra(KeyCode.News.NEWS_ID,id);
        context.startActivity(intent);
    }
    private TextView mTxtNewsTitle;
    private TextView mTxtNewsContent;

    @Override
    public int onBindLayout() {
        return R.layout.activity_news_detail;
    }

    @Override
    public void initView() {
        mTxtNewsTitle = findViewById(R.id.txt_news_detail_title);
        mTxtNewsContent = findViewById(R.id.txt_news_detail_content);
    }

    @Override
    public void initData() {
        int newsid = getIntent().getIntExtra(KeyCode.News.NEWS_ID,-1);
        mViewModel.getNewsDetailById(newsid);
    }

    public void showNewsDetail(NewsDetail newsDetail) {
        mTxtNewsTitle.setText(newsDetail.getTitle());
        mTxtNewsContent.setText(newsDetail.getContent());
    }

    @Override
    public Class<NewsDetailViewModel> onBindViewModel() {
        return NewsDetailViewModel.class;
    }

    @Override
    public ViewModelProvider.Factory onBindViewModelFactory() {
        return NewsViewModelFactory.getInstance(getApplication());
    }

    @Override
    public void initViewObservable() {
        mViewModel.getNewsDetailSingleLiveEvent().observe(this, new Observer<NewsDetail>() {
            @Override
            public void onChanged(@Nullable NewsDetail newsDetail) {
                showNewsDetail(newsDetail);
            }
        });
    }
}

File: module_news/src/main/java/com/fly/tour/news/NewsDetailActivity.kt
package com.fly.tour.news

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.fly.tour.common.base.BaseMvpActivity
import com.fly.tour.common.event.KeyCode
import com.fly.tour.db.entity.NewsDetail
import com.fly.tour.news.contract.NewsDetailContract
import com.fly.tour.news.model.NewsDetailModel
import com.fly.tour.news.presenter.NewsDetailPresenter
import kotlinx.android.synthetic.main.activity_news_detail.*

class NewsDetailActivity :
    BaseMvpActivity<NewsDetailModel, NewsDetailContract.View, NewsDetailPresenter>(),
    NewsDetailContract.View {
    companion object {
        fun startNewsDetailActivity(context: Context, id: Int) {
            val intent = Intent(context, NewsDetailActivity::class.java)
            intent.putExtra(KeyCode.News.NEWS_ID, id)
            context.startActivity(intent)
        }
    }

    override fun showViewDetail(newsDetail: NewsDetail?) {
        mTxtNewsDetailTitle.text = newsDetail?.title
        mTxtNewsDetailContent.text = newsDetail?.content
    }

    override fun initPresenter(): NewsDetailPresenter {
        return NewsDetailPresenter(this)
    }

    override fun onBindLayout(): Int {
        return R.layout.activity_news_detail
    }

    override fun initData() {
        var newsid = intent.getIntExtra(KeyCode.News.NEWS_ID, -1)
        mPresenter?.getNewsDetailById(newsid)
    }

}
